src/core/agent/services/signifyNotificationService.ts

```diff
@@ -57,7 +58,10 @@ class SignifyNotificationService extends AgentService {
 
       const notifications = await this.signifyClient
         .notifications()
-        .list(startFetchingIndex, startFetchingIndex + 24);
+        .list(startFetchingIndex, startFetchingIndex + 24)
+        .catch(() => {
+          Agent.agent.bootAndConnect();
+        });
```

Comment:
I think we can `await` here, right? Because we want the loop to stop trying to fetch notifications until it's ready again.

Reply:
I just call it here to make the Agent.isOnline to `false`. It will still throw an error after that and we will re-initialize everything when isOnline is true again.

Reply:
But I want it to:
- Poll for KERIA notifications
- Stop if it goes offline
- Start polling again once KERIA comes back online

So if we `await` this should be back online, and we can add a `continue` so that it goes back to the start of the loop to call it again

Reply:
but if we use `await` it will hang until the connection is back and we can't use `continue` inside the `catch`

Reply:
I want the loop to hang - the loop is no good unless the connection is there. The rest of the app won't hang if we call `await`, only the loop will. The current code will continuously call `bootAndConnect` on every iteration of the loop, so there will be a constantly growing number of recursive bootAndConnect calls.

We can't do this - there should just be one call to `bootAndConnect` at any given time, or we will be calling `boot()` hundreds of times a second.

Reply:
And why can't we use `continue` inside a catch block? JS allows this

Reply:
With the current code, I didn't return anything in the `catch` block then the `notifications` will be `undefined`, so the loop won't need to be hang, it won't continue the iteration because it will exit the method (tomorrow I will add a `throw` under this line to make things clearer). Then we can initialize the loop again from the `AppWrapper` based on the `isOnline`. The `continue` won't work if we put it inside the `catch` here because it comes inside the promise, it will work if we use a try/catch block.

Reply:
Follow up question which is even more important:
```
        Agent.agent.connections.onConnectionStateChanged((event) => {
          return connectionStateChangedHandler(event, dispatch);
        });
        Agent.agent.signifyNotifications.onNotificationStateChanged((event) => {
          return keriaNotificationsChangeHandler(event, dispatch);
        });
        Agent.agent.credentials.onAcdcStateChanged((event) => {
          return acdcChangeHandler(event, dispatch);
        });
```

If KERIA goes offline and online 5 times - and then we receive accept/admit a new ACDC (`onAcdcStateChanged`) - will there be 5 instances of `connectionStateChangedHandler` being called?

Same goes for the Signify Notifications - if it goes offline and online very quickly in between calls to KERIA (so this error is never thrown in `signifyNotifications.ts` - will we end up with multiple instances of signifyNotifications polling KERIA? It feels possible if (signifyNotification calls KERIA OK, KERIA goes offline, user clicks identifier card details, KERIA comes online after, bootAndConnect connects, next loop of signifyNotification calls KERIA OK).

Reply:
So I'm not sure if we should even be restarting these event handlers - e.g. `onAcdcStateChanged` is used to mark credential as pending or confirmed locally - it isn't really directly related to KERIA.

So I would be thinking we set these up _once_ on start-up and that's it; and we make signifyNotification.ts able to handle KERIA going down without losing the loop, with `await` and this:
```
if (notifications === undefined) {
  // KERIA went down while querying, is now back online
  continue;
}
```

Reply:
Yeah, you are right, I just tested again, looks like it will create another instance. I will change it.